atc/configvalidate/validate.go

```diff
@@ -386,3 +393,23 @@ func validateVarSources(c Config) ([]ConfigWarning, error) {
 
 	return warnings, compositeErr(errorMessages)
 }
+
+func validateDisplay(c Config) ([]ConfigWarning, error) {
+	var warnings []ConfigWarning
+
+	if c.Display == nil {
+		return warnings, nil
+	}
+
+	url, err := url.Parse(c.Display.BackgroundImage)
+
+	if err != nil {
+		return warnings, fmt.Errorf("background_image is not a valid URL: %s", c.Display.BackgroundImage)
+	}
+
+	if url.Scheme == "data" {
```

Comment:
Okay, so I'm just being paranoid, but *maybe* we should be explicit about what schemes are valid (i.e. allow `http`/`https`/no scheme for relative URLs, rather than deny `data`).

Any sane browser will do this validation for us, but from what I can tell, certain older versions of IE are vulnerable to an XSS attack via the `javascript` scheme: https://owasp.org/www-community/xss-filter-evasion-cheatsheet#style-tag-using-background-image

Probably overkill, I know, but I figure it's better to add schemes in the future if people want them rather than trying to retroactively patch a vulnerability (that probably doesn't exist, but still) - or worse still, have the (unlikely) vulnerability go unnoticed by the "good guys"

Feel free to tell me I'm being crazy 😁  if you think this is totally unnecessary, lemme know

Reply:
No I think that is reasonable - I was basing denying data off of what would work to show and image in that tag (from here https://developer.mozilla.org/en-US/docs/Web/CSS/url()) but handling the case of someone trying to do something naughty is also useful. Should also provide a bit of validation that the scheme is one that `url()` understands so may help users if they put in a URL with a scheme that wouldn't work (e.g. `tcp://`).

I've amended the last validation commit with that.